### DISCUSSION TITLE
Does the CSI driver automatically detach volumes from nodes if it goes down and schedules it on another node

### DISCUSSION THREAD

**Discussion by cryptobench** [vitobotta/hetzner-k3s#360] [Q&A] [answered]
I'm looking to build a service that might be needing more performant storage than Longhorn RWX for some usecases, so I was looking to understand if the current Hetzner-CSI driver will allow me to attach a volume to a node, let kubernetes schedule a pod on that node which for example could be a database, and if that said node goes down then the CSI driver would automatically detach the volume from the dead node and attach it to a new node and schedule the database on then ew node alongside the same volume?

**Comment by vitobotta** [comment, marked as answer]
Yes, that's the default behaviour although there is some time delay before Kubernetes decides the node is not coming back up and therefore it attempts to reattach the volume on another node. 

A side note: while performance with Hetzner volumes is pretty good (around 300MB/sec sequential both read and write, and around 7500 IOPS both read and write), if you need the highest performance for databases I would recommend you use the local path provisioner so to use the local disk directly.

**Reply by vitobotta** [reply]
Forgot to mention that that would of course work well if there is replication at application level, which is typical with databases in HA configuration.

**Reply by cryptobench** [reply]
So if i'm using a local path provisioner, would that mean that I should have some nodes specifically for high performance and only use those for that sort of requests from my users? Or what's the defacto standard approach to this problem?

**Reply by vitobotta** [reply]
Local storage on cloud instance is very fast NVME storage, so that's plenty of IOPS for databases. So shared compute instances are fine. Of course if you want more predictable performance also on the compute side, you can use instances with dedicated cores.

**Reply by cryptobench** [reply]
Predictability of performance is nothing that i'm worrying about. I'm more looking to understand for example in cases where there is no replication at the application level, how can I make sure that the correct volume is then scheduled to the correct node if using local path provisioner?

**Reply by vitobotta** [reply]
If there is no replication at application level I don't recommend using the local path provisioner.

**Reply by cryptobench** [reply]
Got it. Thanks for the tips! Much appreciated :-)

**Reply by vitobotta** [reply]
np :)
